projects/hello_world/src/hello_world.c

```diff
@@ -0,0 +1,35 @@
+//
+// Created by henry on 2021-01-09.
+//
+#include <stdint.h>
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+void callback(SoftTimerId id, void *context) {
+  static bool b_increment = false;
+  Counters *Counter = context;
+  LOG_DEBUG("%d\n", ++Counter->counter_a);
+  if (b_increment) {
+    LOG_DEBUG("%d\n", ++Counter->counter_b);
+    b_increment = false;
+  } else {
+    b_increment = true;
+  }
+}
+int main(void) {
+  Counters Counter = { 0 };
+  interrupt_init();
+  soft_timer_init();
+  while (true) {
+    if(!soft_timer_inuse()){
+      soft_timer_start(0.5 * 1000000, callback, &Counter, NULL);
+    }
```

Comment:
This is definitely the most creative solution to this homework I've seen :) The solution hinted at by the homework is to start one soft timer in main, then at the end of the soft timer callback, start *another* soft timer to form a loop. That way you don't have to constantly poll here, and you can just use a plain old `while (true) {}` and let the soft timers do their thing.

Other tips:
* you can use `soft_timer_start_millis` (see soft_timer.h) to avoid having to multiply by 1000000, you can just specify the duration in milliseconds
* it's good practice to `#define` a constant for the duration of the soft timer and use that instead of the magic numbers here - make sure the constant has a descriptive name and mentions the unit it's in (i.e. ms or us)
* I see Travis CI (our automatic build running CI system) failed because the `if` line isn't formatted correctly - you can use `make format` to autoformat our code in accordance with our style.